src/test/unit/parameter_group.ld

```diff
@@ -0,0 +1,10 @@
+SECTIONS {
+  .pg_registry :   SUBALIGN(4)  /* force 4-byte alignment for pg_regisry. Gcc default is 32 bytes, causing padding insertion and breaking the init structure. */
+  {
+    PROVIDE_HIDDEN (__pg_registry_start = .);
+    KEEP (*(.pg_registry))
+    KEEP (*(SORT(.pg_registry.*)))
+    PROVIDE_HIDDEN (__pg_registry_end = .);
+  }
+}
+INSERT AFTER .fini_array;
```

Comment:
appending it to .fini should work. I just tried to keep it in same place as ARM build, but anything in .text will do

crtbegin.o or crt0.o is probably placement-sensitive